src/test/java/org/elasticsearch/indices/store/IndicesStoreIntegrationTests.java

```diff
@@ -95,6 +95,12 @@ public void indexCleanup() throws Exception {
         assertThat(Files.exists(indexDirectory(node_3, "test")), equalTo(false));
 
         logger.info("--> move shard from node_1 to node_3, and wait for relocation to finish");
+        SlowClusterStateProcessing disruption = null;
+        if (randomBoolean()) {
+            disruption = new SlowClusterStateProcessing(node_3, getRandom(), 0, 0, 1000, 2000);
```

Comment:
Oh i see now what you do. You want to make sure the shard is deleted even if CS processing is slow (or but not missing). I think its ok now. It will be hard to make this work with an absolute block as it will also be in the way of the relocation command.